src/components/modals/baseModal.vue

```diff
@@ -0,0 +1,37 @@
+<template>
+  <Teleport to="#modals"
+    ><UseFocusTrap :options="options">
+      <div
+        class="fixed inset-0 bg-gray-900 bg-opacity-40"
+        ref="target"
+        @click="dismissModal"
+      >
+        <div
+          class="fixed inset-0 flex items-center justify-center w-auto m-auto max-w-[500px]"
+        >
+          <div class="p-4 bg-white text-black rounded-md w-full" @click.stop>
+            <slot></slot>
+          </div>
+        </div></div
+    ></UseFocusTrap>
+  </Teleport>
+</template>
+<script setup lang="ts">
+import { ref } from 'vue';
+import { UseFocusTrap } from '@vueuse/integrations/useFocusTrap/component';
+const target = ref(null);
+const options = {
+  immediate: true,
+  onDeactivate: () => {
+    emit('dismiss', false);
+  },
+};
+
+const emit = defineEmits<{
+  (e: 'dismiss', value: boolean): void;
+}>();
+
+const dismissModal = () => {
+  emit('dismiss', false);
+};
```

Comment:
you could also add a test on baseModal to test that it works correctly and also focus trap works.